src/go/rdctl/pkg/utils/utils.go

```diff
@@ -0,0 +1,77 @@
+package utils
+
+import (
+	"io/fs"
+	"os"
+	"path"
+
+	"github.com/rancher-sandbox/rancher-desktop/src/go/rdctl/pkg/directories"
+)
+
+// Get the parent (or grandparent, or great-grandparent...) directory of fullPath.
+// numberTimes is the number of steps to ascend in the directory hierarchy.
+func MoveToParent(fullPath string, numberTimes int) string {
+	fullPath = path.Clean(fullPath)
+	for ; numberTimes > 0; numberTimes-- {
+		fullPath = path.Dir(fullPath)
+	}
+	return fullPath
+}
+
+/**
+ * Verify the path exists. For Linux pass in mode bits to guarantee the file is executable (for at least one
+ * category of user). Note that on macOS the candidate is a directory, so never pass in mode bits.
+ * And mode bits don't make sense on Windows.
+ */
+func CheckExistence(candidatePath string, modeBits fs.FileMode) string {
+	stat, err := os.Stat(candidatePath)
+	if err != nil {
+		return ""
+	}
+	if modeBits != 0 && (!stat.Mode().IsRegular() || stat.Mode().Perm()&modeBits == 0) {
+		// The modeBits check is only for executability -- we only care if at least one of the three
+		// `x` mode bits is on. So this check isn't used for a general permission-mode-bit check.
+		return ""
+	}
+	return candidatePath
+}
+
+// Returns the absolute path to the Rancher Desktop executable.
+// Returns an empty string if the executable was not found.
+func GetWindowsRDPath(rdctlPath string) string {
```

Comment:
No need to do it now, but I wonder if it would make sense to move the various `get*RDPath` methods all into `utils` with the same name (with per-platform implementations).

Reply:
I thought about that too. I came to the same conclusion, that it was better left for another PR. It is an attractive modification to #3808...